### PR TITLE
final set of type annotation

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

File: smac/epm/rf_with_instances.py [hunk 1]
```diff
@@ -132,8 +131,6 @@ def __init__(
                        n_points_per_tree, ratio_features, min_samples_split,
                        min_samples_leaf, max_depth, eps_purity, self.seed]
 
-        self.logger = logging.getLogger(self.__module__ + "." + self.__class__.__name__)
```
[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/epm/rf_with_instances.py]: The base class creates a logger, which is a picklable subclass from [the utils module](https://github.com/automl/SMAC3/blob/master/smac/utils/logging.py). This statement assigns the standard, non-picklable logger instead. I don't know why this assignment ended up here, but the logger is correctly initialized in the base class so we don't need this here.

File: smac/utils/validate.py [hunk 2]
```diff
@@ -182,19 +182,21 @@ def validate(self,
         runs, validated_rh = self._get_runs(config_mode, instance_mode, repetitions, runhistory)
 
         # Create new Stats without limits
-        inf_scen = Scenario({'run_obj': self.scen.run_obj,
-                             'cutoff_time': self.scen.cutoff, 'output_dir': ""})
+        inf_scen = Scenario({
+            'run_obj': self.scen.run_obj,
+            'cutoff_time': self.scen.cutoff,  # type: ignore[attr-defined] # noqa F821
+            'output_dir': ""})
```
[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/utils/validate.py]: Unfortunately, I don't know how this function works. Maybe @shukon can help here?

File: smac/utils/validate.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def validate(self,
                  backend: str = 'threading',
                  runhistory: RunHistory = None,
                  tae: ExecuteTARun = None,
-                 output_fn: str = "",
+                 output_fn: typing.Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/utils/validate.py]: I will check and fix this.

[the remark this answers is not in the document]
Comment (mfeurer) [reply smac/utils/validate.py]: After briefly looking at this, I'm wondering about the `if output_fn` statement and the statement which was there before. It basically prevented from going into `_save_results`. So either the check in `_save_results` or the check in the validate function is unnecessary. I prefer to drop the checks in the validate function(s).

[the remark this answers is not in the document]
Comment (KEggensperger) [reply smac/utils/validate.py]: ok

**Pull request merged**